packages/waku/src/router/client.ts

```diff
@@ -301,8 +305,19 @@ const InnerRouter = ({ routerData }: { routerData: RouterData }) => {
   const refetch = useRefetch();
 
   const [route, setRoute] = useState(() =>
-    parseRoute(new URL(window.location.href)),
+    // This is the first initialization of the route, and it has
+    // to ignore the hash, because on server side there is none.
+    // Otherwise there will be a hydration error.
+    // The client side route, including the hash, will be updated in the effect below.
+    parseRoute(new URL(window.location.href), true),
   );
+
+  // Update the route post-load to include the current hash.
+  useEffect(() => {
+    setRoute(parseRoute(new URL(window.location.href)));
```

Comment:
This actually triggers re-renders every time, even if the hash is `''`. We should probably improve it.